stacks-signer/src/signer.rs

```diff
@@ -1380,8 +1455,27 @@ impl Signer {
                     self.approved_aggregate_public_key
                 );
             }
+            if matches!(self.state, State::OperationInProgress(Operation::Dkg)) {
+                // We already have DKG, abort operation and reset state.
+                debug!(
+                    "{self}: DKG has already been set. Aborting DKG round {}.",
+                    self.coordinator.current_dkg_id
+                );
+                self.finish_operation();
+            }
```

Comment:
Can you explain this code a little bit? I have no reason to doubt it, I'm just not totally following this code path

Reply:
Yeah this is because I was working on a different PR simultaneously (see https://github.com/stacks-network/stacks-core/pull/4660) It is one of the reasons I converted it to a draft so I can rebase against it once merged :) 

Basically this is to handle the fact that we could trigger a DKG round due to timing issues where we might have an outdated view (for example, exceeded the threshold check but still not gotten the approved key though the other PR will mitigate this). If this happens and then we later discover that DKG was triggered and it shouldn't have been, we should abort our attempt. This is incomplete however until we can then load the appropriate DKG shares from storage for the winning round. Hence the link to Marten's issue. So I have put this ticket a bit on pause until I can build off these two other PRs.